src/test/specs/generic/consensus/base/block/BlockInterlink.spec.js

```diff
@@ -0,0 +1,52 @@
+describe('BlockInterlink', () => {
+    const hash1 = new Hash(BufferUtils.fromBase64(Dummy.hash1));
+    const hash2 = new Hash(BufferUtils.fromBase64(Dummy.hash2));
+
+    const blockHashes = [hash1, hash2];
+    const blockInterlink1 = new BlockInterlink(blockHashes);
+
+    it('must have a well defined blockHashes array', () => {
+        /* eslint-disable no-unused-vars */
+        expect(() => {
+            const test1 = new BlockInterlink(undefined);
+        }).toThrow('Malformed blockHashes');
+
+        expect(() => {
+            const test1 = new BlockInterlink(null);
+        }).toThrow('Malformed blockHashes');
+
+        expect(() => {
+            const test1 = new BlockInterlink(1);
+        }).toThrow('Malformed blockHashes');
+
+        expect(() => {
+            const test1 = new BlockInterlink(new Uint8Array(101));
+        }).toThrow('Malformed blockHashes');
+        /* eslint-enable no-unused-vars */
+    });
+
+    it('is serializable and unserializable', () => {
+        const blockInterlink2 = BlockInterlink.unserialize(blockInterlink1.serialize());
+        expect(BufferUtils.equals(blockInterlink1, blockInterlink2)).toBe(true);
+        expect(BufferUtils.equals(blockInterlink1.hash(), blockInterlink2.hash())).toBe(true);
```

Comment:
`(await blockInterlink1.hash()).equals(await blockInterlink2.hash())` + add async function in test